### PR TITLE
[Backend] TRT backend & PP-Infer backend support pinned memory

### PR DESCRIPTION
**Pull request by wang-xinyu**
### PR types(PR类型)

Backend

### Describe

- Introduced [pinned host memory](https://developer.nvidia.com/blog/how-optimize-data-transfers-cuda-cc/)
- Add enable_pinned_memory in Runtime Option
- Add a new buffer type `FDDeviceHostBuffer`
- Integrate `FDDeviceHostBuffer` into `FDTensor`
- TRT backend optimization with `FDDeviceHostBuffer`
- Paddle Inference backend support pinned memory option

### CODE REVIEW

**Comment by wang-xinyu** [comment]
After integrating pinned memory, the inference latency is comparable with trtexec. It mainly optimized the D2H latency.

The latency includes H2D, GPU inference and D2H.

Tested on P40.

The reason why yolov6s has only a little improvement is the output tensor shape of yolov6s(8400x85) is much smaller than yolov5s(2x25200x85) and yolov7(2x25200x85).

|model| before | after | trtexec|
|:--|:--|:--|:--|
|yolov5s | 11.0566ms | 7.34001ms | 6.76131ms|
|yolov6s |  7.17801ms | 6.483ms | 5.69616ms |
|yolov7 | 25.7099ms | 19.9956ms | 19.8108ms |

File: fastdeploy/core/fd_tensor.cc [hunk 1]
```diff
@@ -227,11 +248,20 @@ void FDTensor::FreeFn() {
 }
 
 void FDTensor::CopyBuffer(void* dst, const void* src, size_t nbytes) {
-  if (device == Device::GPU) {
+  if (is_pinned_memory) {
```
Comment (jiangjiajun) [inline fastdeploy/core/fd_tensor.cc]: `is_pinned_memory`仍然是CPU内存，那看起来应该放在Device的判断之后，跟现有的CPU的代码逻辑在同一个地方

File: fastdeploy/core/fd_tensor.cc [hunk 2]
```diff
@@ -191,6 +191,23 @@ void FDTensor::PrintInfo(const std::string& prefix) {
 }
 
 bool FDTensor::ReallocFn(size_t nbytes) {
+  if (is_pinned_memory) {
```
Comment (jiangjiajun) [inline fastdeploy/core/fd_tensor.cc]: 同`CopyBuffer`函数的处理逻辑

**Review by jiangjiajun — CHANGES_REQUESTED** [review]

File: fastdeploy/core/fd_tensor.cc (hunk 1 above)
Comment (wang-xinyu) [reply fastdeploy/core/fd_tensor.cc]: fixed

File: fastdeploy/core/fd_tensor.cc (hunk 2 above)
Comment (wang-xinyu) [reply fastdeploy/core/fd_tensor.cc]: fixed

File: fastdeploy/runtime.cc [hunk 3]
```diff
@@ -356,6 +356,10 @@ void RuntimeOption::EnableTrtFP16() { trt_enable_fp16 = true; }
 
 void RuntimeOption::DisableTrtFP16() { trt_enable_fp16 = false; }
 
+void RuntimeOption::EnableTrtPinnedMemory() { trt_enable_pinned_memory = true; }
```
Comment (jiangjiajun) [inline fastdeploy/runtime.cc]: 接口改为`EnablePinedMemory()`  (对应里面的`trt_enable_pinned_memory`，也改为`enable_pinned_memory`)

**Review by jiangjiajun — CHANGES_REQUESTED** [review]

File: fastdeploy/runtime.cc (hunk 3 above)
Comment (wang-xinyu) [reply fastdeploy/runtime.cc]: Done

**Comment by wang-xinyu** [comment]
Also verified pinned memory in Paddle Inference GPU backend, tested on PPYoloE. The latency is almost the same as before. Because the output size is small(7200 + 4).

**Review by jiangjiajun — APPROVED** [review]

**Pull request merged**